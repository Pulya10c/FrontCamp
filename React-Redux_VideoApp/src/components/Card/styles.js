import styled from 'styled-components';
import { Link } from 'react-router-dom';

export const CardStyled = styled(Link)`
  display: grid;
  grid-gap: 12px;
  align-content: start;
  padding: 4px;
  font-family: 'Open Sans', sans-serif;
  border-radius: 10px;
  background-color: #fff;
  border: 1px solid rgba(218, 220, 224, 0.5);
  box-shadow: 0 20px 20px -20px $main-color;
  transition: all 0.3s cubic-bezier(0.075, 0.82, 0.165, 1);
  cursor: pointer;
  &:hover {
    transform: scale(1.02);
    box-shadow: 0 2px 2px #00183a;
  }
`;
