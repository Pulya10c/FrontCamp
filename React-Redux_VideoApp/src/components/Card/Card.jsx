import React from "react";
import PropTypes from "prop-types";

import { CardStyled } from "./styles";

const Card = ({ posterPath, title, year, subTitle, to, description }) => {
  return (
    <CardStyled to={to}>
      <img src={posterPath} alt={title} />
      <h4>{title}</h4>
      <span>{subTitle}</span>
      <p>{description}</p>
      <p>{year}</p>
    </CardStyled>
  );
};

Card.propTypes = {
  posterPath: PropTypes.string,
  title: PropTypes.string,
  year: PropTypes.string,
  subTitle: PropTypes.string,
  to: PropTypes.string,
  description: PropTypes.string
};

export default Card;
