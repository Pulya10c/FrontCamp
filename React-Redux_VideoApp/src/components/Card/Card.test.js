import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

describe('Component Card', () => {
  let wrapper, defaultProps;

  beforeEach(() => {
    defaultProps = {
      posterPath: 'mockPosterPath',
      title: 'mockTitle',
      year: 'mockYear',
      subTitle: 'mockSubTitle',
      to: 'mockTo',
      description: 'mockDescription',
    };
    wrapper = shallow(<Card {...defaultProps} />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
