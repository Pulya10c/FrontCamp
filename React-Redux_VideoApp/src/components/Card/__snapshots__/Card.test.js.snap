// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Card should render correctly 1`] = `
<Styled(Link)
  to="mockTo"
>
  <img
    alt="mockTitle"
    src="mockPosterPath"
  />
  <h4>
    mockTitle
  </h4>
  <span>
    mockSubTitle
  </span>
  <p>
    mockDescription
  </p>
  <p>
    mockYear
  </p>
</Styled(Link)>
`;
